dbt/include/clickhouse/macros/materializations/distributed_table.sql

```diff
@@ -0,0 +1,101 @@
+    {%- set preexisting_backup_relation = load_cached_relation(backup_relation) -%}
+    {% if not existing_relation.can_exchange %}
+      {%- set intermediate_relation =  make_intermediate_relation(target_relation_local) -%}
+      {%- set preexisting_intermediate_relation = load_cached_relation(intermediate_relation) -%}
+    {% endif %}
+  {% endif %}
+  {% set view_relation = default__make_temp_relation(target_relation, '__dbt_tmp') %}
+  -- drop the temp relations if they exist already in the database
+  {{ drop_relation_if_exists(preexisting_intermediate_relation) }}
+  {{ drop_relation_if_exists(preexisting_backup_relation) }}
+  {{ drop_relation_if_exists(view_relation) }}
+
+  {% set grant_config = config.get('grants') %}
+
+  {{ run_hooks(pre_hooks, inside_transaction=False) }}
+
+  {% call statement('main') %}
+    {{ create_view_as(view_relation, sql) }}
+  {% endcall %}
+
+  {{ run_hooks(pre_hooks, inside_transaction=True) }}
+
+  {% if backup_relation is none %}
+    {% do run_query(create_empty_table_from_view(target_relation_local, view_relation)) or '' %}
+    {% do run_query(create_distributed_table(target_relation, target_relation_local)) or '' %}
+  {% elif existing_relation.can_exchange %}
+    -- We can do an atomic exchange, so no need for an intermediate
+    {% call statement('main') -%}
+      {% do run_query(create_empty_table_from_view(backup_relation, view_relation)) or '' %}
+    {%- endcall %}
+    {% do exchange_tables_atomic(backup_relation, existing_relation) %}
+  {% else %}
+    {% do run_query(create_empty_table_from_view(intermediate_relation, view_relation)) or '' %}
+    {{ adapter.rename_relation(existing_relation_local, backup_relation) }}
+    {{ adapter.rename_relation(intermediate_relation, target_relation_local) }}
+  {% endif %}
+  {% do run_query(clickhouse__insert_into(target_relation, sql)) or '' %}
+  {{ drop_relation_if_exists(view_relation) }}
+  -- cleanup
+  {% set should_revoke = should_revoke(existing_relation, full_refresh_mode=True) %}
+  {% do apply_grants(target_relation, grant_config, should_revoke=should_revoke) %}
+
+  {% do persist_docs(target_relation, model) %}
+  {{ run_hooks(post_hooks, inside_transaction=True) }}
+  {{ adapter.commit() }}
+  {{ drop_relation_if_exists(backup_relation) }}
+  {{ run_hooks(post_hooks, inside_transaction=False) }}
+  {{ return({'relations': [target_relation]}) }}
+
+{% endmaterialization %}
+
+{% macro create_distributed_table(relation, local_relation) %}
+   {%- set cluster = adapter.get_clickhouse_cluster_name()[1:-1] -%}
+   {%- set sharding = config.get('sharding_key') -%}
+
+    CREATE TABLE {{ relation }} {{ on_cluster_clause() }}
+    ENGINE = Distributed('{{ cluster}}', '{{ relation.schema }}', '{{ local_relation.name }}'
+    {% if sharding is none %}
+        , {{ sharding }}
+    {% endif %}
+    )
+    AS SELECT * FROM {{ local_relation}} LIMIT 0
+ {% endmacro %}
+
+{% macro create_empty_table_from_view(relation, view_relation) -%}
```

Comment:
Maybe rename this to "create_empty_table_from_relation", since it's not specific to a view.

Reply:
Yes, renamed